### ISSUE TITLE
protractor tests for ngStrap alerts

### ISSUE THREAD

**Issue by jshemas** [mgcrea/angular-strap#1396]
For the life of me I can not figure out how to write a protractor test for these alerts. What i'm trying to test is when someone clicks a button, the alert shows. I want to make sure it has the right text inside...

Something I've tried is... (after the user clicks the button)

var el = $('.alert');
browser.driver.wait(protractor.until.elementIsVisible(el));

Then after the alert is on the page I would do something like...

 expect($('.alert span')).toEqual('look at this message!');

But because the alert markup doesn't exist on the page, Protractor can't find it.

TL;DR: How can I write a Protractor to wait till the alert is on the page, then test its text?

**Comment by jshemas** [comment]
Still haven't been able to find a fix for this. =/

**Comment by dziamid** [comment]
I've run into this as well. The issue is not the presence of the element on the page, but the blocking behavior of the alert. If you run protractor with `--elementExplorer` you should see that when alert in being shown, all commands to protractor timeout .. until an alert is hidden.

**Comment by jshemas** [comment]
@dziamid Thanks for the info! Unfortunately, I ended up building my own alert system that relays alert container all ready existing on the page.

**Comment by dziamid** [comment]
Why close? The issue is still not resolved.

**Comment by alsfurlan** [comment]
I am getting this problem now. Anyone can help me?
